core/src/test/java/org/testcontainers/DaemonTest.java

```diff
@@ -0,0 +1,56 @@
+package org.testcontainers;
+
+import org.junit.Test;
+import org.rnorth.visibleassertions.VisibleAssertions;
+import org.testcontainers.containers.GenericContainer;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class DaemonTest {
+
+    public static void main(String[] args) {
+        Thread mainThread = Thread.currentThread();
+
+        GenericContainer genericContainer = null;
+
+        try {
+            genericContainer = new GenericContainer().withCommand("top");
+            genericContainer.start();
+
+            Set<Thread> threads = new HashSet<>(Thread.getAllStackTraces().keySet());
+            threads.remove(mainThread);
+
+            Set<Thread> nonDaemonThreads = threads.stream().filter(it -> !it.isDaemon()).collect(Collectors.toSet());
+
+            if (nonDaemonThreads.isEmpty()) {
+                VisibleAssertions.pass("All threads marked as daemon");
+            } else {
+                String nonDaemonThreadNames = nonDaemonThreads.stream()
+                    .map(Thread::getName)
+                    .collect(Collectors.joining("\n", "\n", ""));
+
+                VisibleAssertions.fail("Expected all threads to be daemons but the following are not:\n" + nonDaemonThreadNames);
+            }
+        } finally {
+            if (genericContainer != null) {
+                genericContainer.stop();
+            }
+        }
+    }
+
+    @Test
+    public void testThatAllThreadsAreDaemons() throws Exception {
+        ProcessBuilder processBuilder = new ProcessBuilder(
```

Comment:
we need to fork the JVM, otherwise we might already initialize the threads and the result will not reflect the scenario ("should not create non-daemon threads")

Reply:
Creative solution 😄